Antidote/CallsManager.m

```diff
@@ -0,0 +1,401 @@
+{
+    [self.manager sendCallControl:OCTToxAVCallControlCancel
+                           toCall:self.currentCall
+                            error:nil];
+}
+
+#pragma mark - RingingCallViewController Delegate
+
+- (void)ringingCallAnswerButtonPressed:(RingingCallViewController *)controller
+{
+    [self.manager answerCall:self.currentCall
+                 enableAudio:YES
+                 enableVideo:NO error:nil];
+}
+
+- (void)ringingCallDeclineButtonPressed:(RingingCallViewController *)controller
+{
+    [self.manager sendCallControl:OCTToxAVCallControlCancel
+                           toCall:self.currentCall
+                            error:nil];
+}
+
+#pragma mark - Private
+
+- (void)notifyOfIncomingCall:(OCTCall *)call
+{
+    if (! [self.currentCallViewController isKindOfClass:[ActiveCallViewController class]]) {
+        // For now we reject call if we are in a dialing or ringing state
+        [self.manager sendCallControl:OCTToxAVCallControlCancel toCall:call error:nil];
+        return;
+    }
+
+    self.pendingIncomingCall = call;
+
+    OCTFriend *friend = [self.pendingIncomingCall.chat.friends firstObject];
+
+    ActiveCallViewController *activeVC = (ActiveCallViewController *)self.currentCallViewController;
+    [activeVC createIncomingCallViewForFriend:friend.nickname];
+}
+
+- (AbstractCallViewController *)viewControllerForCall:(OCTCall *)call
+{
+    AbstractCallViewController *viewController;
+
+    OCTFriend *friend = [call.chat.friends firstObject];
+
+    switch (call.status) {
+        case OCTCallStatusActive: {
+            ActiveCallViewController *activeVC = [ActiveCallViewController new];
+            activeVC.delegate = self;
+            viewController = activeVC;
+            break;
+        }
+        case OCTCallStatusDialing: {
+            DialingCallViewController *dialingVC = [DialingCallViewController new];
+            dialingVC.delegate = self;
+            viewController = dialingVC;
+            break;
+        }
+        case OCTCallStatusRinging: {
+            RingingCallViewController *ringingVC = [RingingCallViewController new];
+            ringingVC.delegate = self;
+            viewController = ringingVC;
+            break;
+        }
+        case OCTCallStatusPaused:
+            NSAssert(NO, @"We should not be here yet. Not yet implemented");
+            break;
+    }
+
+    viewController.nickname = friend.nickname;
+    viewController.modalInPopover = YES;
+    viewController.modalPresentationStyle = UIModalPresentationOverCurrentContext;
+
+    return viewController;
+}
+
+- (void)dismissAndSwitchViewControllerForCall:(OCTCall *)call
+{
+    AbstractCallViewController *abstractVC = [self viewControllerForCall:call];
+
+    [self.callNavigation setViewControllers:@[abstractVC] animated:YES];
+
+    self.currentCallViewController = abstractVC;
+
+    self.currentCall = call;
+}
+
+- (void)dismissAndSwitchToActiveViewControllerForCall:(OCTCall *)call
```

Comment:
I'm not sure if we need separate method for that. You can move this logic to `switchViewControllerForCall:`.

This way you can simple call `switch...` and doesn't care if you switched to active or non-active view controller